lib/esp32/espnow.toit

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+STATION ::= 0 // Wi-Fi station mode
+SOFTAP  ::= 1 // Not support yet
+
+BROADCAST_MAC ::= #[0xff, 0xff, 0xff, 0xff, 0xff, 0xff]
+
+check_mac_ address/ByteArray:
+  if address.size != 6:
+      throw "ESP-Now MAC address length must be 6 bytes"
+
+check_key_ key/ByteArray:
+  if key.size != 16:
+      throw "ESP-Now key length must be 16 bytes"
+
+check_channel_ channel/int:
+  if channel > 14 or channel < 0:
+      throw "ESP-Now channel range must be 0~14"
+
+class Address:
+  address/ByteArray
+
+  constructor .address:
+    check_mac_ address
```

Comment:
The `check_mac_` should be inlined here, and all other methods should use the `Address` class.